### PR TITLE
Replace assertions with value errors on distilbert model

### PR DESCRIPTION
**Pull request by JuheonChu**
# Replace assertions with ValueErros on distilbert model

This PR is made to check if the valid checks made from #20433 pass for all cases.

Co-author: @batese2001

To: @younesbelkada

### CODE REVIEW

File: src/transformers/models/distilbert/modeling_distilbert.py [hunk 1]
```diff
@@ -810,7 +822,9 @@ def __init__(self, config: PretrainedConfig):
 
         self.distilbert = DistilBertModel(config)
         self.qa_outputs = nn.Linear(config.dim, config.num_labels)
-        assert config.num_labels == 2
+        if config.num_labels != 2:
+            raise ValueError(f"config.num_labels should be 2, but it is {(config.num_labels)}")
```
Comment (younesbelkada) [inline src/transformers/models/distilbert/modeling_distilbert.py:826]:
```suggestion
            raise ValueError(f"config.num_labels should be 2, but it is {config.num_labels}")
```

**Review by younesbelkada — COMMENTED** [review]
Thanks a lot for your PR! I made a small suggestion and it should be ready for merging ;) 
EDIT: let me push the small modif and I'll approve the PR once the tests are green!

File: src/transformers/models/distilbert/modeling_distilbert.py [hunk 2]
```diff
@@ -141,7 +142,10 @@ def __init__(self, config: PretrainedConfig):
         self.dim = config.dim
         self.dropout = nn.Dropout(p=config.attention_dropout)
 
-        assert self.dim % self.n_heads == 0
+        # Have an even number of multi heads that divide the dimensions
+        if self.dim % self.n_heads != 0:
+            # Raise value errors for even multi-head attention nodes
+            raise ValueError(f"self.n_heads: {self.n_heads} must divide self.dim: {str(self.dim)} evenly")
```
Comment (younesbelkada) [inline src/transformers/models/distilbert/modeling_distilbert.py:148]:
```suggestion
            raise ValueError(f"self.n_heads: {self.n_heads} must divide self.dim: {self.dim} evenly")
```

**Comment by JuheonChu** [comment]
Thank you so much!

**Review by younesbelkada — APPROVED** [review]
Thank you very much for taking care of refactoring `distilbert` modeling file and making the assertions cleaner!
Looking forward to your next contributions on the ecosystem ;) 
Leaving this PR now to @sgugger for final approval!

**Comment by JuheonChu** [comment]
Thank you so much for your guidance and help!

File: src/transformers/models/distilbert/modeling_distilbert.py (hunk 2 above)
Comment (JuheonChu) [reply src/transformers/models/distilbert/modeling_distilbert.py]: Hello @younesbelkada, may I suggest to change the comment on line 147 with "Raise ValueErrors for **uneven** multi-head attention nodes"

File: src/transformers/models/distilbert/modeling_distilbert.py [hunk 3]
```diff
@@ -141,7 +142,10 @@ def __init__(self, config: PretrainedConfig):
         self.dim = config.dim
         self.dropout = nn.Dropout(p=config.attention_dropout)
 
-        assert self.dim % self.n_heads == 0
+        # Have an even number of multi heads that divide the dimensions
+        if self.dim % self.n_heads != 0:
+            # Raise value errors for even multi-head attention nodes
```
Comment (JuheonChu) [inline src/transformers/models/distilbert/modeling_distilbert.py:147]: May I ask you if I can change this comment saying "Raise ValueErrors for uneven multi-head attention nodes"?

Comment (younesbelkada) [reply src/transformers/models/distilbert/modeling_distilbert.py:147]: Sure! Go for it (no need to open a new PR ;) )

**Review by sgugger — APPROVED** [review]
Thanks a lot!

**Pull request merged**